src/ray/raylet/node_manager.cc

```diff
@@ -2051,37 +2045,6 @@ void NodeManager::ScheduleTasks(
   RAY_CHECK(local_queues_.GetTasks(TaskState::PLACEABLE).size() == 0);
 }
 
-void NodeManager::TreatTaskAsFailed(const Task &task, const ErrorType &error_type) {
```

Comment:
I wanted to remove this method because we really shouldn't be using it anymore - the owner should decide if a task has failed, not a raylet. The raylet can still use [MarkObjectsAsFailed](https://github.com/ray-project/ray/blob/master/src/ray/raylet/node_manager.cc#L2085) if it knows the ObjectIDs to fail. If it really does need to fail a task/resource request (although I can't really think of any cases where it would), it should return the right error to the owner.